api/apps/api/src/modules/scenarios/planning-units/scenario-planning-units-protection-status-calculator-service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { DbConnections } from '@marxan-api/ormconfig.connections';
+import { ScenariosPlanningUnitGeoEntity } from '@marxan/scenarios-planning-unit';
+import { Scenario } from '../scenario.api.entity';
+import { isNil } from 'lodash';
+
+@Injectable()
+export class ScenarioPlanningUnitsProtectedStatusCalculatorService {
+  constructor(
+    @InjectRepository(
+      ScenariosPlanningUnitGeoEntity,
+      DbConnections.geoprocessingDB,
+    )
+    private readonly puRepo: Repository<ScenariosPlanningUnitGeoEntity>,
+  ) {}
+
+  /**
+   * Calculate default lock status of planning units given their intersection
+   * with protected areas.
+   */
+  async calculatedProtectionStatusForPlanningUnitsIn(scenario: Scenario): Promise<void> {
+    /**
+     * If no protected areas are defined, or no threshold has been defined,
+     * there's nothing to do. We allow to set these on update, so we should not
+     * enforce this here.
```

Comment:
Could be a private method like `shouldSetDefaultLockingStatus` to express this comment.

Reply:
done in 51fb372c282d7b5244bf5b2c7004e3eee37d4349

Reply:
(left some comments in any case because memory 🤷🏼)